ext/docker/runmount_builder_test.go

```
@@ -0,0 +1,141 @@
+	assert.Empty(t, err)
+}
+
+func TestRunmountBuild_setupTempDir(t *testing.T) {
+	dirCreated, err := setupTempDir()
+	assert.Empty(t, err)
+	_, err = os.Stat(dirCreated)
+	assert.True(t, err == nil)
+}
+
+func TestRunmountBuild_createMountContainer(t *testing.T) {
+	sh, ctl := shell.NewTestShell()
+	_, cctl := ctl.CmdFor("docker", "create")
+	cctl.ResultSuccess(fmt.Sprintf("%s", testContainerBuildID), "")
+	ctx := sous.BuildContext{
+		Sh: sh,
+	}
+
+	containerID, err := createMountContainer(ctx, testBuildID)
+	assert.Empty(t, err)
+	assert.Equal(t, testContainerBuildID, containerID)
+}
+
+func getTestRunSpec() MultiImageRunSpec {
+	testRunSpec := MultiImageRunSpec{}
+	specF, _ := os.Open("./testdata/runmountbuilder/run_spec.json")
+	dec := json.NewDecoder(specF)
+	dec.Decode(&testRunSpec)
+
+	return testRunSpec
+}
+
+func TestRunmountBuild_extractRunspec(t *testing.T) {
+	tempDir := "./testdata/runmountbuilder"
+	testRunSpec := getTestRunSpec()
+	sh, ctl := shell.NewTestShell()
+	_, cctl := ctl.CmdFor("docker", "cp")
+	cctl.ResultSuccess("", "")
+	ctx := sous.BuildContext{
+		Sh: sh,
+	}
+
+	runSpec, _ := extractRunSpec(ctx, tempDir, testContainerBuildID)
+	fmt.Println("runspec : ", runSpec)
+
+	assert.Equal(t, testRunSpec, runSpec)
+}
+
+func TestRunmountBuild_validateRunSpec(t *testing.T) {
+	testRunSpec := getTestRunSpec()
+	err := validateRunSpec(testRunSpec)
+	assert.NoError(t, err)
+}
+
+func TestRunmountBuild_constructImageBuilders(t *testing.T) {
+	testRunSpec := getTestRunSpec()
+	builders, err := constructImageBuilders(testRunSpec)
+
+	assert.NoError(t, err)
+	builder := *builders[0]
+	assert.Equal(t, "docker.otenv.com/sous-otj-run", builder.RunSpec.Image.From)
+}
+
+func TestRunmountBuild_extractFiles(t *testing.T) {
+	testRunSpec := getTestRunSpec()
+	sh, ctl := shell.NewTestShell()
+	_, cctl := ctl.CmdFor("docker", "cp")
+	cctl.ResultSuccess("", "")
+	ctx := sous.BuildContext{
+		Sh: sh,
+	}
+	builders, _ := constructImageBuilders(testRunSpec)
```

I don't think we should ignore this error, if it's somehow not nil in this simple case we really need to know.

---

I have just realised that `constructImageBuilders` always returns a nil error, therefore instead that second result parameter should be removed to avoid all these underscores.